### ISSUE TITLE
Calling multiple services with different status codes

### ISSUE THREAD

**Issue by Difreg** [meeshkan/unmock-js#406]
### Description
Within my application I want to cover a function with an integration test, which calls several services one after the other. In this example, I want to test the error handling if one of these requests returns a status code other than 200. 

### Steps to Reproduce
During the test both used services are set to the needed state. Then both calls are sent. (see minimal reproducable example)

### Expected Result 
The first call delivers the response with the status code of the first service (200, {Hello: "World"}), while the second call delivers the response with the corresponding status code of the second service (401, "Unauthorized").

### Actual Result 
When requesting exampleService1, a response is returned with status code 401. Apparently the statuses of all services are changed to the highest.

### Additional Context

Minimal reproducable example:
```js
import unmock, { Service, transform } from "unmock";
import Axios from "axios";
const { withCodes } = transform;

let exampleService1: Service;
let exampleService2: Service;

unmock
  .nock("https://www.example.com", "ExampleService1")
  .get("/exampleEndpoint1")
  .reply(200, { Hello: "World" })
  .get("/example")
  .reply(401, "Unauthorized");

unmock
  .nock("https://www.example.cloud.com", "ExampleService2")
  .get("/exampleEndpoint2")
  .reply(200, { Bella: "Ciao" })
  .get("/example2")
  .reply(401, "Unauthorized");

beforeAll(() => {
  const { ExampleService1, ExampleService2 } = unmock.on().services;
  exampleService1 = ExampleService1;
  exampleService2 = ExampleService2;
});
beforeEach(() => {
  exampleService1.reset();
  exampleService2.reset();
});
afterEach(() => {
  exampleService1.spy.resetHistory();
  exampleService2.spy.resetHistory();
});
afterAll(() => {
  unmock.off();
});

it("example", async () => {
  exampleService1.state(withCodes(200));
  exampleService2.state(withCodes(401));
  // Szenario:    Call ExampleService 1
  // Expectation: Call succeeds (State.Codes = 200)
  // Reality:     Call fails with status code 401
  await Axios.get("https://www.example.com/exampleEndpoint1")
    .then(e => {
      console.log(e.data);
    })
    .catch(e => {
      console.log(e);
    });
});
```

Bug isolated in cooperation with @jonasgrunert

**Comment by mikesol** [comment]
Thanks for reporting this. I can confirm that it's a bug. I will investigate over the weekend, and I'll try to have a release out by Monday. I'll also write a test to address this specific bug and will reference it in the changelog.

**Comment by Difreg** [comment]
Awesome! Thank you very much :)

**Comment by mikesol** [comment]
I have submitted https://github.com/meeshkan/unmock-js/pull/407, which fixes two bugs that your example triggered. Thanks for reporting it!

**Comment by Difreg** [comment]
Awesome! Would love to see the pull request approved :p

**Comment by mikesol** [comment]
Apologies for the delay and thanks for the reminder! The PR was merged and a new version with the fix, 0.3.17, was just published.
